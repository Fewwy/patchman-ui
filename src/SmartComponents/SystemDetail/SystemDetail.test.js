import SystemDetail from './SystemDetail';
import { useLocation } from 'react-router-dom';
import ShallowRenderer from 'react-test-renderer/shallow';

jest.mock('react-redux', () => ({
    ...jest.requireActual('react-redux'),
    useSelector: () => ({ id: 'entity-id' })
}));

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useLocation: jest.fn(() => ({ state: 'advisories' }))
}));
const renderer = new ShallowRenderer();

describe('SystemDetail.js', () => {
    it('Should match the snapshots', () => {
        renderer.render(
            <SystemDetail />
        );
        const result = renderer.getRenderOutput();
        expect(result).toMatchSnapshot();
    });

    it('Should match the snapshot when Package tab is active by default', () => {
        useLocation.mockImplementation(() => ({ state: { tab: 'packages' } }));
        renderer.render(
            <SystemDetail />
        );
        const result = renderer.getRenderOutput();
        expect(result).toMatchSnapshot();
    });
});

