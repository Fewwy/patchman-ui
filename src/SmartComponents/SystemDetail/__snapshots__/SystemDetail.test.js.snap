// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SystemDetail.js Should match the snapshot when Package tab is active by default 1`] = `
<Tabs
  activeKey={0}
  className="patchTabSelect"
  component="div"
  hasBorderBottom={true}
  isBox={false}
  isFilled={false}
  isHidden={true}
  isSecondary={false}
  isVertical={false}
  leftScrollAriaLabel="Scroll left"
  mountOnEnter={false}
  onSelect={[Function]}
  onToggle={[Function]}
  ouiaSafe={true}
  rightScrollAriaLabel="Scroll right"
  unmountOnExit={false}
  variant="default"
>
  <Tab
    data-ouia-component-id="system-advisories-tab"
    data-ouia-component-type="system-advisories-tab"
    eventKey={0}
    title={
      <TabTitleText>
        Advisories
      </TabTitleText>
    }
  >
    <SystemAdvisories
      handleNoSystemData={[Function]}
    />
  </Tab>
  <Tab
    data-ouia-component-id="system-packages-tab"
    data-ouia-component-type="system-packages-tab"
    eventKey={1}
    title={
      <TabTitleText>
        Packages
      </TabTitleText>
    }
  >
    <SystemPackages
      handleNoSystemData={[Function]}
    />
  </Tab>
</Tabs>
`;

exports[`SystemDetail.js Should match the snapshots 1`] = `
<Tabs
  activeKey={0}
  className="patchTabSelect"
  component="div"
  hasBorderBottom={true}
  isBox={false}
  isFilled={false}
  isHidden={true}
  isSecondary={false}
  isVertical={false}
  leftScrollAriaLabel="Scroll left"
  mountOnEnter={false}
  onSelect={[Function]}
  onToggle={[Function]}
  ouiaSafe={true}
  rightScrollAriaLabel="Scroll right"
  unmountOnExit={false}
  variant="default"
>
  <Tab
    data-ouia-component-id="system-advisories-tab"
    data-ouia-component-type="system-advisories-tab"
    eventKey={0}
    title={
      <TabTitleText>
        Advisories
      </TabTitleText>
    }
  >
    <SystemAdvisories
      handleNoSystemData={[Function]}
    />
  </Tab>
  <Tab
    data-ouia-component-id="system-packages-tab"
    data-ouia-component-type="system-packages-tab"
    eventKey={1}
    title={
      <TabTitleText>
        Packages
      </TabTitleText>
    }
  >
    <SystemPackages
      handleNoSystemData={[Function]}
    />
  </Tab>
</Tabs>
`;
